### PR TITLE
Verticle line through table

### PR DESCRIPTION
**Pull request by tyro668**
table height cannot be grid height, it will through the table.

### CODE REVIEW

**Comment by weisJ** [comment]
Could you please elaborate what precisely the issue being fixed is here?

**Comment by tyro668** [comment]
> 

if use JTable inside with a scrollpane,the table‘s vertical lines are not end at the last row,in fact the vertical tables grid lines though the last row,like this image:

https://github.com/tyro668/pictures/blob/main/Xnip2022-09-08_10-45-34.png

**Comment by weisJ** [comment]
The way the vertical lines are painted is intentional. If you think there should be an option to change that behaviour, please file a feature request issue. I highly appreciate any contributions, but would like them to be discussed beforehand if they do more than simply changing something trivial (like translations etc.). In this case I think adding an option to to change the painting behaviour through a client property would be very reasonable (similar to `TableConstants.KEY_ALTERNATE_ROW_COLOR `).

Note that
1. The height of the cells doesn't need to be computed as it is already available above as the bottom coordinate of `maxCell`.
2. As is your patch doesn't work when dragging columns

**Comment by tyro668** [comment]
Fix the problem,add property:TableConstants.KEY_VERTICAL_LINES_THROUGH

File: core/src/main/java/com/github/weisj/darklaf/ui/table/TableConstants.java [hunk 1]
```diff
@@ -42,6 +42,7 @@ public interface TableConstants extends CellConstants {
     String KEY_IS_FILE_LIST = "Table.isFileList";
     String KEY_IS_PRINT_MODE = "Table.printMode";
     String KEY_CELL_VALUE_DETERMINES_EDITOR_CLASS = "Table.cellDeterminesClass";
+    String KEY_VERTICAL_LINES_THROUGH = "throughVerticalLines";
```
Comment (weisJ) [inline core/src/main/java/com/github/weisj/darklaf/ui/table/TableConstants.java:45]:
```suggestion
    String KEY_FULL_HEIGHT_VERTICAL_LINES = "Table.fullHeightVerticalLines";
```

File: core/src/main/java/com/github/weisj/darklaf/ui/table/DarkTableUI.java [hunk 2]
```diff
@@ -274,7 +274,7 @@ protected void paintGrid(final Graphics g, final int rMin, final int rMax, final
         boolean scrollVisible = scrollBarVisible();
         if (table.getShowVerticalLines()) {
             TableColumnModel cm = table.getColumnModel();
-            int tableHeight = table.getHeight();
+            int tableHeight = PropertyUtil.getBooleanProperty(table, KEY_VERTICAL_LINES_THROUGH, false) ? maxCell.y + maxCell.height - minCell.y : table.getHeight();
```
Comment (weisJ) [inline core/src/main/java/com/github/weisj/darklaf/ui/table/DarkTableUI.java:277]:
```suggestion
            int tableHeight = PropertyUtil.getBooleanProperty(table, KEY_FULL_HEIGHT_VERTICAL_LINES, false) ? maxCell.y + maxCell.height - minCell.y : table.getHeight();
```

File: core/src/main/java/com/github/weisj/darklaf/ui/table/DarkTableUI.java [hunk 3]
```diff
@@ -394,7 +394,7 @@ protected void paintDraggedArea(final Graphics g, final int rMin, final int rMax
             int x1 = vacatedColumnRect.x;
             int y1 = 0;
             int x2 = x1 + vacatedColumnRect.width - 1;
-            int y2 = y1 + tableHeight;
+            int y2 = PropertyUtil.getBooleanProperty(table, KEY_VERTICAL_LINES_THROUGH, false) ? maxCell.y + maxCell.height - minCell.y : y1 + tableHeight;
```
Comment (weisJ) [inline core/src/main/java/com/github/weisj/darklaf/ui/table/DarkTableUI.java:397]:
```suggestion
            int y2 = PropertyUtil.getBooleanProperty(table, KEY_FULL_HEIGHT_VERTICAL_LINES, false) ? maxCell.y + maxCell.height - minCell.y : y1 + tableHeight;
```

**Review by weisJ — CHANGES_REQUESTED** [review]
If we go forward with this change, then the value should be initialized in `installDefaults` by using a value exposed in `table.properties`.

Besides that I was wondering whether https://docs.oracle.com/javase/7/docs/api/javax/swing/JTable.html#setFillsViewportHeight(boolean) would solve your particular issue. In this case the fay to move forward would be to respect the value set by the user.

**Pull request closed without merging**